packages/next-auth/src/core/lib/assert.ts

```diff
@@ -94,6 +99,20 @@ export function assertConfig(params: {
   let hasTwitterOAuth2
 
   for (const provider of options.providers) {
+    if (provider.type === "oauth" && !provider.issuer) {
```

Comment:
```suggestion
    if (provider.type === "oauth" && (!provider.wellKnown || !provider.issuer)) {
```

`wellKnown` should provide the endpoint configs. This check currently breaks providers such as Azure AD who rely on `wellKnown`.